hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientPartitionServiceImpl.java

```diff
@@ -130,10 +127,7 @@ private boolean getPartitions() {
             Future<ClientMessage> future = getPartitionsFrom(connection);
             ClientMessage responseMessage = future.get();
             ClientGetPartitionsCodec.ResponseParameters response = ClientGetPartitionsCodec.decodeResponse(responseMessage);
-            if (response == null) {
-                return false;
-            }
-            return processPartitionResponse(response);
+            return response != null && processPartitionResponse(response);
```

Comment:
Idea suggests all crazy things. Don't listen it . I also prefer the older version.